app/views/common/os_selection/_pxe_loader.html.erb

```diff
@@ -0,0 +1,10 @@
+<%= fields_for item do |f| %>
+  <% selected_attr = f.object.new_record? && item.hostgroup.pxe_loader.nil? ? { :selected => @operatingsystem.preferred_loader } : {} -%>
+  <%= selectable_f f, :pxe_loader,
+    @operatingsystem.available_loaders.to_a, {
+      :include_blank => blank_or_inherit_f(f, :pxe_loader),
+      :selected => item.pxe_loader
+    }.merge(selected_attr),
+    :label => _('PXE Loader'),
+    :help_inline => popover(_("DHCP filename option"), _("TFTP filename to use, select one of")  + ' ' + @operatingsystem.available_loaders.values.join(', ') + _('or set to blank for Non-PXE method (e.g. iPXE)'), :data => { :placement => 'top' }) %>
+<% end %>
```

Comment:
It works! The only missing workflow is when I select a hostgroup without loader set (sane default is preselected) and then I select a hostgroup without loader set. It is ignored, but not big issue IMHO.